File: pog-store/core/src/main/java/br/com/pogstore/core/event/EventStream.java
package br.com.pogstore.core.event;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

public class EventStream {

	private UUID eventSourceId;
	private final List<Event> events;
	private long startVersion;
	private long lastVersion;
	private final Boolean isEmpty;

	public EventStream(final UUID eventsourceId, final List<Event> events,
			final long fromVersion, final long toVersion) {
		this.eventSourceId = eventsourceId;
		this.events = events;
		this.startVersion = fromVersion;
		this.lastVersion = toVersion;
		this.isEmpty = false;
	}

	private EventStream() {
		this.events = new ArrayList<>();
		isEmpty = true;
	}

	public UUID getEventSourceId() {
		return eventSourceId;
	}

	public List<Event> getEvents() {
		return events;
	}

	public long getStartVersion() {
		return startVersion;
	}

	public long getLastVersion() {
		return lastVersion;
	}

	public static EventStream empty() {
		return new EventStream();
	}

	public Boolean isEmpty() {
		return isEmpty;
	}

}

File: cqrs/crime-map/src/CrimeMap.Core/Events/EventStream.cs
using System;
using System.Collections.Generic;

namespace CrimeMap.Core.Events {

	public class EventStream {

		public Guid EventSourceId { get; private set; }

		public IEnumerable<IEvent> Events { get; private set; }

		public long StartVersion { get; private set; }

		public long LastVersion { get; private set; }

		private bool _empty;

		public EventStream(Guid eventsourceId, IEnumerable<IEvent> events, long fromVersion, long toVersion) {
			EventSourceId = eventsourceId;
			Events = events;
			StartVersion = fromVersion;
			LastVersion = toVersion;
			_empty = false;
		}

		private EventStream() {
			Events = new List<IEvent>();
			_empty = true;
		}

		public static EventStream Empty() {
			return new EventStream();
		}

		public bool IsEmpty() {
			return _empty;
		}

	}

}
